File: src/rl_dda_demo_back/api/routes/agents.py
from fastapi import APIRouter

router = APIRouter(prefix="/agents", tags=["agents"])


@router.get("")
async def list_agents() -> list[dict]:
    # Static list for v1; can be moved to DB later
    return [
        {"id": "agent-beginner", "skill": "beginner", "model_version": "v1", "description": "초급 에이전트"},
        {"id": "agent-intermediate", "skill": "intermediate", "model_version": "v1", "description": "중급 에이전트"},
        {"id": "agent-advanced", "skill": "advanced", "model_version": "v1", "description": "고급 에이전트"},
    ]

File: worker/routes/agents.ts
import { Hono } from 'hono';

const agentsRouter = new Hono();

agentsRouter.get('', (c) => {
  // 정적 에이전트 목록 (나중에 DB로 이동 가능)
  return c.json([
    {
      id: 'agent-beginner',
      skill: 'beginner',
      model_version: 'v1',
      description: '초급 에이전트',
    },
    {
      id: 'agent-intermediate',
      skill: 'intermediate',
      model_version: 'v1',
      description: '중급 에이전트',
    },
    {
      id: 'agent-advanced',
      skill: 'advanced',
      model_version: 'v1',
      description: '고급 에이전트',
    },
  ]);
});

export default agentsRouter;

